Feature/4

valerachuk opened the pull request.

valerachuk committed:
login form + jwt auth + auth guards

valerachuk committed:
form fix

delete !important;
add "singe-line" and delete "solo" attribute;
add "Open Sans" font;
move styles to common.css;

valerachuk committed:
rename "token" to "accessToken"

rename "token" to "accessToken" in vuex (JWT)

valerachuk committed:
fix server error handle

valerachuk committed:
Merge branch 'dev' into feature/4

valerachuk committed:
Merge branch 'dev' into feature/4

horba commented on front/src/styles/common.css:
```diff
@@ -1,3 +1,60 @@
+  user-select: none;
+}
+
+.auth-form{
```
all css rules related to login form should be placed in styles of login view

horba commented on front/src/views/LoginCard.vue:
```diff
@@ -0,0 +1,97 @@
+<template>
```
it is really big view code, espetially when you add styles, so it is more compact to separate template, js code and styles

horba commented on front/src/styles/common.css:
```diff
@@ -1,3 +1,60 @@
+.input-group > label {
+  margin: 0 0 5px 18px;
+  font-size: 12px;
+  line-height: 125%;
```
why do you use percentage?

valerachuk replied:
Just copied this property from figma...
Should I delete it ?

horba commented on front/src/styles/common.css:
```diff
@@ -1,3 +1,60 @@
+@import url('https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,300;0,400;0,600;0,700;0,800;1,300;1,400;1,600;1,700;1,800&display=swap');
+
+.auth-card {
```
if default view of components needs changes to be up to design and these changes will be actual for view of this component in different views it is better to place in common rules to prevent duplication of css

valerachuk replied:
it is already in common.css ...

horba commented on front/src/styles/common.css:
```diff
@@ -1,3 +1,60 @@
+  font-weight: 600;
+  font-size: 14px;
+  cursor: pointer;
+  color: #273B7A;
```
ira left link in channel how to customise error text color

horba commented on front/src/styles/common.css:
```diff
@@ -1,3 +1,60 @@
+}
+
+.input-group > .label-error {
+  color: #F44336;
```
please, check if it is actual after theme changes in dev

it seems like it is common rules not only for login

if it is some type of primary color it should be created css variables in variables.css and use it in other places

horba commented on front/src/main.js:
```diff
@@ -3,6 +3,7 @@ import App from './App.vue';
 import vuetify from './plugins/vuetify';
+import axios from 'axios';
 
 import Default from './layouts/Default.vue';
```
please, add @layout path and use it like

import { Default, Authorize } from '@layout'

horba commented on front/src/router/index.js:
```diff
@@ -19,7 +19,8 @@ import {
-  MyLeisure
+  MyLeisure,
+  LoginCard
 } from '@views';
```
please, change import one component per line onto several to save vertical space in component

valerachuk committed:
prettied

add @layout to alias
rename Login to SignIn
separate SignIn to different files
add mmt-text-input component
line-heigth % --> em

valerachuk committed:
rename files
